### ISSUE TITLE
Drupal 10, PHP 8.2 and PHPStan issues 

### ISSUE THREAD

**Issue by millnut** [localgovdrupal/localgov_directories#315]
The work on https://github.com/localgovdrupal/localgov_project/pull/131 has highlighted some PHP 8.2 and PHPStan issues which need fixing.

```
 ------ --------------------------------------------------------------------------------------------------------- 
  Line   modules/contrib/localgov_directories/modules/localgov_directories_location/src/ProximitySearchSetup.php  
 ------ --------------------------------------------------------------------------------------------------------- 
  75     Variable $index_datasrc in empty() always exists and is not falsy.                                       
 ------ --------------------------------------------------------------------------------------------------------- 

 ------ -------------------------------------------------------------------------------------------------------------------------------- 
  Line   modules/contrib/localgov_directories/modules/localgov_directories_venue/tests/src/Kernel/ViewUpgradeForProximitySearchTest.php  
 ------ -------------------------------------------------------------------------------------------------------------------------------- 
  42     Variable $display_for_proximity_search in empty() always exists and is not falsy.                                               
 ------ -------------------------------------------------------------------------------------------------------------------------------- 

 ------ ------------------------------------------------------------------ 
  Line   modules/contrib/localgov_directories/src/ConfigurationHelper.php  
 ------ ------------------------------------------------------------------ 
  218    Variable $config_src in empty() always exists and is not falsy.   
 ------ ------------------------------------------------------------------ 

 ------ ---------------------------------------------------------------------------------------------------------------------------------- 
  Line   modules/contrib/localgov_directories/src/Form/LocalgovDirectoriesFacetsForm.php                                                   
 ------ ---------------------------------------------------------------------------------------------------------------------------------- 
  53     Method Drupal\localgov_directories\Form\LocalgovDirectoriesFacetsForm::save() should return int but return statement is missing.  
 ------ ---------------------------------------------------------------------------------------------------------------------------------- 

 ------ -------------------------------------------------------------------------------------------------------------------------------------- 
  Line   modules/contrib/localgov_directories/src/Form/LocalgovDirectoriesFacetsTypeForm.php                                                   
 ------ -------------------------------------------------------------------------------------------------------------------------------------- 
  95     Variable $message might not be defined.                                                                                               
  97     Method Drupal\localgov_directories\Form\LocalgovDirectoriesFacetsTypeForm::save() should return int but return statement is missing.  
 ------ -------------------------------------------------------------------------------------------------------------------------------------- 

 ------ -------------------------------------------------------------------------------------------------------------------------------- 
  Line   modules/contrib/localgov_directories/tests/src/Functional/ResetFacetFilterTest.php                                              
 ------ -------------------------------------------------------------------------------------------------------------------------------- 
  74     Access to an undefined property Drupal\Tests\localgov_directories\Functional\ResetFacetFilterTest::$directory_channel_node_id.  
         💡 Learn more: https://phpstan.org/blog/solving-phpstan-access-to-undefined-property                                            
  103    Access to an undefined property Drupal\Tests\localgov_directories\Functional\ResetFacetFilterTest::$directory_channel_node_id.  
         💡 Learn more: https://phpstan.org/blog/solving-phpstan-access-to-undefined-property                                            
  132    Access to an undefined property Drupal\Tests\localgov_directories\Functional\ResetFacetFilterTest::$directory_channel_node_id.  
         💡 Learn more: https://phpstan.org/blog/solving-phpstan-access-to-undefined-property                                            
 ------ -------------------------------------------------------------------------------------------------------------------------------- 

 ------ ---------------------------------------------------------------------------------------------------------------------------------- 
  Line   modules/contrib/localgov_directories/tests/src/Kernel/EntityReferenceChannelsSelectionTest.php                                    
 ------ ---------------------------------------------------------------------------------------------------------------------------------- 
  82     Access to an undefined property Drupal\Tests\localgov_directories\Kernel\EntityReferenceChannelsSelectionTest::$directory_nodes.  
         💡 Learn more: https://phpstan.org/blog/solving-phpstan-access-to-undefined-property                                              
  87     Access to an undefined property Drupal\Tests\localgov_directories\Kernel\EntityReferenceChannelsSelectionTest::$page_type.        
         💡 Learn more: https://phpstan.org/blog/solving-phpstan-access-to-undefined-property                                              
  97     Access to an undefined property Drupal\Tests\localgov_directories\Kernel\EntityReferenceChannelsSelectionTest::$other_type.       
         💡 Learn more: https://phpstan.org/blog/solving-phpstan-access-to-undefined-property                                              
  113    Access to an undefined property Drupal\Tests\localgov_directories\Kernel\EntityReferenceChannelsSelectionTest::$page_type.        
         💡 Learn more: https://phpstan.org/blog/solving-phpstan-access-to-undefined-property                                              
  114    Access to an undefined property Drupal\Tests\localgov_directories\Kernel\EntityReferenceChannelsSelectionTest::$page_type.        
         💡 Learn more: https://phpstan.org/blog/solving-phpstan-access-to-undefined-property                                              
  119    Access to an undefined property Drupal\Tests\localgov_directories\Kernel\EntityReferenceChannelsSelectionTest::$directory_nodes.  
         💡 Learn more: https://phpstan.org/blog/solving-phpstan-access-to-undefined-property                                              
  120    Access to an undefined property Drupal\Tests\localgov_directories\Kernel\EntityReferenceChannelsSelectionTest::$directory_nodes.  
         💡 Learn more: https://phpstan.org/blog/solving-phpstan-access-to-undefined-property                                              
  125    Access to an undefined property Drupal\Tests\localgov_directories\Kernel\EntityReferenceChannelsSelectionTest::$directory_nodes.  
         💡 Learn more: https://phpstan.org/blog/solving-phpstan-access-to-undefined-property                                              
  126    Access to an undefined property Drupal\Tests\localgov_directories\Kernel\EntityReferenceChannelsSelectionTest::$other_type.       
         💡 Learn more: https://phpstan.org/blog/solving-phpstan-access-to-undefined-property                                              
  131    Access to an undefined property Drupal\Tests\localgov_directories\Kernel\EntityReferenceChannelsSelectionTest::$directory_nodes.  
         💡 Learn more: https://phpstan.org/blog/solving-phpstan-access-to-undefined-property                                              
  132    Access to an undefined property Drupal\Tests\localgov_directories\Kernel\EntityReferenceChannelsSelectionTest::$directory_nodes.  
         💡 Learn more: https://phpstan.org/blog/solving-phpstan-access-to-undefined-property                                              
  137    Access to an undefined property Drupal\Tests\localgov_directories\Kernel\EntityReferenceChannelsSelectionTest::$other_type.       
         💡 Learn more: https://phpstan.org/blog/solving-phpstan-access-to-undefined-property                                              
  138    Access to an undefined property Drupal\Tests\localgov_directories\Kernel\EntityReferenceChannelsSelectionTest::$other_type.       
         💡 Learn more: https://phpstan.org/blog/solving-phpstan-access-to-undefined-property                                              
  141    Access to an undefined property Drupal\Tests\localgov_directories\Kernel\EntityReferenceChannelsSelectionTest::$directory_nodes.  
         💡 Learn more: https://phpstan.org/blog/solving-phpstan-access-to-undefined-property                                              
  144    Access to an undefined property Drupal\Tests\localgov_directories\Kernel\EntityReferenceChannelsSelectionTest::$directory_nodes.  
         💡 Learn more: https://phpstan.org/blog/solving-phpstan-access-to-undefined-property                                              
  145    Access to an undefined property Drupal\Tests\localgov_directories\Kernel\EntityReferenceChannelsSelectionTest::$directory_nodes.  
         💡 Learn more: https://phpstan.org/blog/solving-phpstan-access-to-undefined-property                                              
 ------ ---------------------------------------------------------------------------------------------------------------------------------- 
```

**Comment by Adnan-cds** [comment]
Hello,
Some of the `Variable $VAR-NAME-GOES-HERE in empty() always exists and is not falsy.` messages above were the product of a phpstan bug which appears to have been [fixed recently](https://github.com/phpstan/phpstan/issues/9403).  Perhaps we need to update our phpstan version.

**Comment by millnut** [comment]
Hi @Adnan-cds the [phpstan used on the action](https://github.com/localgovdrupal/localgov_project/actions/runs/6316343354/job/17150728755#step:9:198) was phpstan/phpstan (1.10.35) which appears to be the latest, can you see a PR at phpstan side that says this is fixed or is waiting for a release?

EDIT: Looking some more they have fixed this on 1.11.x but have not created any releases for 1.11.x yet. Let's ignore errors of that type for now.

**Comment by millnut** [comment]
Hey @Adnan-cds I did some testing with the latest 1.10.x and 1.11.x releases of PHPStan and those errors still occur despite the fix in PHPStan, I'll refactor slightly the same as I did in https://github.com/localgovdrupal/localgov_guides/pull/130 for the check to pass
